### DISCUSSION TITLE
Out of sync PR generates destructiveChanges

### DISCUSSION THREAD

**Discussion by sjurgis** [scolladon/sfdx-git-delta#688] [Q&A]
Day 3 of going live with new pipeline and I've found something I haven't accounted for, haha!

Person A and B create a feature branch each from latest develop
person A added work
person B added work
person A merged their PR
person B opens PR - sgd generated destructiveChanges with A's work (yikes)

I think quick fix would be pipeline merging in latest develop, but that doesn't feel right and likely prone to other problems.

My script is 

```
    git fetch origin $BITBUCKET_PR_DESTINATION_BRANCH
    sfdx sgd:source:delta --from origin/$BITBUCKET_PR_DESTINATION_BRANCH \
    --to $BITBUCKET_COMMIT \
    --ignore .forceignore $DELTA \
    --output deploy
```

**Comment by scolladon** [comment]
Hi @sjurgis !

As the B branch is out of sync with `develop` there are multiple approach here:

- Do not run automation until the PR branch is in sync with `develop` (merge `develop` or rebase on top of new `develop` HEAD)
- Run sgd against the [common ancestor](https://github.com/scolladon/sfdx-git-delta#:~:text=from%20a%20common%20ancestor). Be careful as in this scenario, the content of the A branch will be ignored. Some component can miss content (classes, triggers, lwc, flow, etc)

**Comment by sjurgis** [comment]
> Do not run automation until the PR branch is in sync

IIRC thats default in github, but I'm using bitbucket pipelines. Any chance you know how?

**Reply by scolladon** [reply]
I have a very limited experience of bitbucket pipelines, sorry...
I tried to search for something like: 'Bitbucket run pipeline branch not synced' without wonderful result

**Reply by sjurgis** [reply]
Found a setting `Maximum number of commits behind destination branch` in `Edit a branch restriction` - that might be it

**Reply by alex-vitry** [reply]
The restriction above in Bitbucket doesn't allow 0 commits behind (to force branches in sync), minimum is 1, go figure. Fyi [https://jira.atlassian.com/browse/BCLOUD-22431](https://jira.atlassian.com/browse/BCLOUD-22431)

**Comment by soileaud** [comment]
@sjurgis  Which option did you end up going with, and how has it fared?

**Reply by sjurgis** [reply]
It's been a while and I left project before it got battle tested, but we've came up with way to get common ancestor
```
    git fetch origin $BITBUCKET_PR_DESTINATION_BRANCH
    pointer=$(git merge-base $BITBUCKET_COMMIT origin/$BITBUCKET_PR_DESTINATION_BRANCH)
    echo 'Common ancestor: '$pointer
    sfdx sgd:source:delta \
    --from $pointer \
    --to $BITBUCKET_COMMIT \
    --ignore .forceignore $DELTA \
    --output deploy
```

Hope I never need to work with bitbucket again (I swear to myself every few years lol).
